include/nccl_ofi_mr.h

```diff
@@ -116,6 +133,7 @@ static inline nccl_ofi_mr_ckey_t nccl_ofi_mr_ckey_mk_dmabuf(int fd, uint64_t off
 
 static inline nccl_ofi_mr_ckey_t nccl_ofi_mr_ckey_mk_vec(void *iov_base, size_t iov_len)
 {
+	nccl_ofi_mr_ckey_round(&iov_len, &iov_base, "iovec");
```

Comment:
Not super comfortable with this.

@bwbarrett is this safe? Previously, the "cache key", a stupid name on my part but which sort-of made sense at the time, was just a container for the arguments given from NCCL. Any internal adjustments for the sake of simplifying the cache (ie: rounding up or down) was internal to the cache code, but never would have changed the at-rest value in the "cache key". This change makes that no longer true -- which means the value passed into fi_mr_regattr is the rounded value instead of the raw value.

Maybe this is fine, given the set of dims that we can potentially get from NCCL?

Reply:
```cpp
static inline void nccl_ofi_mr_ckey_fill_mr_attrs(nccl_ofi_mr_ckey_ref ckey, struct fi_mr_attr *attrs, uint64_t *flags)
{
	assert(ckey->type != NCCL_OFI_MR_CKEY_INVALID);
	*flags = 0;
#if HAVE_DECL_FI_MR_DMABUF
	if (ckey->type == NCCL_OFI_MR_CKEY_DMABUF) {
		*flags |= FI_MR_DMABUF;
	}
	attrs->dmabuf = (const struct fi_mr_dmabuf *)ckey;
#endif
	attrs->mr_iov = (const struct iovec *)ckey;
	attrs->iov_count = 1;
}
```

This is relying on punning with `mr_iov/dmabuf` in fi_mr_regattr:

```cpp
struct fi_mr_attr {
    union {
        const struct iovec *mr_iov;
        const struct fi_mr_dmabuf *dmabuf;
    };
    size_t             iov_count;
    uint64_t           access;
    uint64_t           offset;
    uint64_t           requested_key;
    void               *context;
    size_t             auth_key_size;
    uint8_t            *auth_key;
    enum fi_hmem_iface iface;
    union {
        uint64_t       reserved;
        int            cuda;
        int            ze
        int            neuron;
        int            synapseai;
    } device;
    void               *hmem_data;
    size_t             page_size;
    const struct fid_mr *base_mr;
    size_t             sub_mr_cnt;
};
```

Reply:
https://github.com/aws/aws-ofi-nccl/pull/707 to make this more obvious when reading the code

Reply:
Mark as resolved per discussion in the slack channel. Thanks!